File: scripts/generate_manifest.py
import os
import json

def generate_manifest():
    data_dir = 'public/data'
    maps = ["AmbroseValley", "GrandRift", "Lockdown"]
    dates = []
    files = []

    # Scan for date folders
    for entry in sorted(os.listdir(data_dir)):
        entry_path = os.path.join(data_dir, entry)
        if os.path.isdir(entry_path) and entry.startswith('February_'):
            # Convert 'February_11' to '2026-02-11' format for the UI
            day = entry.split('_')[1]
            date_str = f"2026-02-{day}"
            dates.append(date_str)

            # Scan for parquet files in the date folder
            for f in os.listdir(entry_path):
                if f.endswith('.parquet'):
                    # Determine mapId from filename
                    map_id = None
                    for m in maps:
                        if m in f:
                            map_id = m
                            break
                    
                    if map_id:
                        files.append({
                            "date": date_str,
                            "mapId": map_id,
                            "path": f"{entry}/{f}"
                        })

    manifest = {
        "maps": maps,
        "dates": sorted(list(set(dates))),
        "files": files
    }

    with open(os.path.join(data_dir, 'manifest.json'), 'w') as f:
        json.dump(manifest, f, indent=2)
    
    print(f"Generated manifest.json with {len(files)} files.")

if __name__ == "__main__":
    generate_manifest()

File: scripts/generate_manifest.js
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';
import { parquetReadObjects } from 'hyparquet';

const __dirname = path.dirname(fileURLToPath(import.meta.url));

async function generateManifest() {
  const dataDir = path.join(__dirname, '../public/data');
  const entries = [];

  const folders = fs.readdirSync(dataDir).filter(f => 
    fs.statSync(path.join(dataDir, f)).isDirectory() && f.startsWith('February_')
  );

  console.log(`Processing ${folders.length} folders...`);

  for (const folder of folders) {
    const folderPath = path.join(dataDir, folder);
    const day = folder.split('_')[1];
    const date = `2026-02-${day.padStart(2, '0')}`;

    const files = fs.readdirSync(folderPath);
    console.log(`- ${folder}: ${files.length} files`);

    for (const file of files) {
      if (file.includes('.nakama') || file.endsWith('.parquet')) {
        const filePath = path.join(folderPath, file);
        
        try {
          const nodeBuffer = fs.readFileSync(filePath);
          const arrayBuffer = nodeBuffer.buffer.slice(nodeBuffer.byteOffset, nodeBuffer.byteOffset + nodeBuffer.byteLength);
          
          const objects = await parquetReadObjects({
            file: arrayBuffer,
            limit: 1
          });

          if (objects.length > 0) {
            const row = objects[0];
            const mapId = row.map_id || "AmbroseValley";
            const userId = row.user_id || file.split('_')[0];
            const matchId = row.match_id ? String(row.match_id).split('.')[0] : file.split('_')[1].split('.')[0];
            let tsNum = 0;
            if (typeof row.ts === 'number') tsNum = row.ts;
            else if (typeof row.ts === 'bigint') tsNum = Number(row.ts);
            else if (row.ts) tsNum = new Date(row.ts).getTime();

            // Handle seconds vs milliseconds
            if (tsNum > 0 && tsNum < 1_000_000_000_000) tsNum *= 1000;

            const entryDate = tsNum > 0 ? new Date(tsNum).toISOString().split('T')[0] : date;

            entries.push({
              date: entryDate,
              folder,
              path: `${folder}/${file}`,
              userId,
              matchId,
              mapId
            });
          }
        } catch (err) {
          console.error(`Error processing ${file}:`, err.message);
        }
      }
    }
  }

  const manifest = {
    generated: new Date().toISOString(),
    totalFiles: entries.length,
    entries: entries
  };

  fs.writeFileSync(path.join(dataDir, 'manifest.json'), JSON.stringify(manifest, null, 2));
  console.log(`\nSuccess! Generated manifest.json with ${entries.length} validated entries.`);
}

generateManifest().catch(console.error);
